nesta_ds_utils/loading_saving/S3.py

```diff
@@ -294,64 +365,85 @@ def download_obj(
     if download_as == "dataframe":
-        if path_from.endswith(tuple([".csv", ".parquet"])):
+        if path_from.endswith(tuple([".csv", ".parquet", ".xlsx", ".xlsm"])):
             return _fileobj_to_df(fileobj, path_from, **kwargs_reading)
         else:
-            raise Exception(
+            raise NotImplementedError(
                 "Download as dataframe currently supported only "
-                "for 'csv' and 'parquet'."
+                "for 'csv','parquet','xlsx' and 'xlsm'."
+            )
+    elif download_as == "geodataframe":
+        if path_from.endswith(tuple([".geojson"])):
+            return _fileobj_to_gdf(fileobj, path_from, **kwargs_reading)
+        else:
+            raise NotImplementedError(
+                "Download as geodataframe currently supported only " "for 'geojson'."
             )
     elif download_as == "dict":
         if path_from.endswith(tuple([".json"])):
             return _fileobj_to_dict(fileobj, path_from, **kwargs_reading)
+        elif path_from.endswith(tuple([".geojson"])):
+            warnings.warn(
+                "Please check geojson has a member with the name 'type', the value of the member must be one of the following:"
+                "'Point', 'MultiPoint', 'LineString', 'MultiLineString', 'Polygon', 'MultiPolygon', 'GeometryCollection',"
+                "'Feature' and 'FeatureCollection'. Else downloaded dictionary will not be valid geojson."
+            )
+            return _fileobj_to_dict(fileobj, path_from, **kwargs_reading)
         else:
-            raise Exception(
-                "Download as dictionary currently supported only " "for 'json'."
+            raise NotImplementedError(
+                "Download as dictionary currently supported only "
+                "for 'json' and 'geojson'."
             )
     elif download_as == "list":
         if path_from.endswith(tuple([".csv", ".txt", ".json"])):
             return _fileobj_to_list(fileobj, path_from, **kwargs_reading)
         else:
-            raise Exception(
+            raise NotImplementedError(
                 "Download as list currently supported only "
                 "for 'csv', 'txt' and 'json'."
             )
     elif download_as == "str":
         if path_from.endswith(tuple([".txt"])):
             return _fileobj_to_str(fileobj)
         else:
-            raise Exception("Download as string currently supported only " "for 'txt'.")
+            raise NotImplementedError(
+                "Download as string currently supported only " "for 'txt'."
+            )
     elif download_as == "np.array":
         if path_from.endswith(tuple([".csv", ".parquet"])):
             return _fileobj_to_np_array(fileobj, path_from, **kwargs_reading)
         else:
-            raise Exception(
+            raise NotImplementedError(
                 "Download as numpy array currently supported only "
                 "for 'csv' and 'parquet'."
             )
     elif not download_as:
```

Comment:
```suggestion
    if not download_as:
```
move this to the top statement